ivy/functional/frontends/tensorflow/general_functions.py

```diff
@@ -347,21 +347,23 @@ def roll(input, shift, axis, name=None):
 
 
 @to_ivy_arrays_and_back
-def split(
-    value, num_or_size_splits, axis=0, num=None, name=None
-):
+def split(value, num_or_size_splits, axis=0, num=None, name=None):
     return ivy.split(
-        value,
-        num_or_size_splits=num_or_size_splits,
-        axis=axis,
-        with_remainder=False
+        value, num_or_size_splits=num_or_size_splits, axis=axis, with_remainder=False
     )
 
 
 def repeat(
-        input,
-        repeats,
-        axis=None,
-        name=None,
+    input,
+    repeats,
+    axis=None,
+    name=None,
 ):
     return ivy.repeat(input, repeats, axis=axis)
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"1.11.0 and below": ("float16", "bfloat16")}, "torch")
+def unstack(value: ivy.array, axis=0, num=None, name=None):
+    res = ivy.unstack(value, axis=axis)
+    return ivy.astype(res, value.dtype)
```

Comment:
You should remove `ivy.astype(res, value.dtype)`

Reply:
If I remove that I get an error for mismatching data type from the ground truth any fixes to solve that?

Reply:
Removed.